### ISSUE TITLE
[Bug]: explicitSpread does not include explicit conditional spreading

### ISSUE THREAD

**Issue by jdwillemse** [jsx-eslint/eslint-plugin-react#3507]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

On occasion I need a property added conditionally. The way I do this is
```jsx
<div {...(actAsBanner && { role: 'banner' })}>Button Link</div>
```
because the following produces invalid HTML since `role` may not be `false`
```jsx
<div role={actAsButton && 'banner'}>Button Link</div>
```

### Expected Behavior

I expect `explicitSpread: "ignore"` to cover this use case since the spreading is completely explicit.

### eslint-plugin-react version

7.31.11

### eslint version

8.28.0

### node version

18.12.1

**Comment by ljharb** [comment]
`role={actAsButton ? 'banner' : undefined}`
